### PR TITLE
Added composite primary key support for update

### PR DESCRIPTION
**Pull request by iNaD**
Hi,

as stated in #79 I discovered that update currently doesn't support composite primary keys (only string based or condition based). I hope I got the static method `id` in the Model class right. As it returns the id which is used for internal storage.

I also added test cases which should cover composite primary keys in the update action.

### CODE REVIEW

File: src/Model.ts [hunk 1]
```diff
@@ -173,7 +173,7 @@ export default class Model {
   }
 
   /**
-   * Gte the value of the primary key.
+   * Get the value of the primary key.
```
Comment (iNaD) [inline src/Model.ts]: Fixed typo

File: src/Model.ts [hunk 2]
```diff
@@ -302,7 +302,7 @@ export default class Model {
   }
 
   /**
-   * Gte Vuex Store insatnce out of connection.
+   * Get Vuex Store insatnce out of connection.
```
Comment (iNaD) [inline src/Model.ts]: same typo as above

File: src/repo/Repo.ts [hunk 3]
```diff
@@ -622,8 +622,11 @@ export default class Repo {
       return
     }
 
-    if (typeof this.entity.primaryKey === 'string') {
-      data[this.entity.primaryKey] && this.query.update(data, data[this.entity.primaryKey])
+    // Use the entities primary key method to retrieve internal id (known as $id)
+    const $id = this.entity.id(data)
```
Comment (iNaD) [inline src/repo/Repo.ts]: I hope I got this one right. If data contains the correct primary key properties a valid id will be generated. This reduces failure when changing the id logic I think.

File: test/feature/RootModules.spec.js [hunk 4]
```diff
@@ -3,12 +3,14 @@ import User from 'test/fixtures/models/User'
 import Profile from 'test/fixtures/models/Profile'
 import Post from 'test/fixtures/models/Post'
 import Comment from 'test/fixtures/models/Comment'
+import CompositeKey from 'test/fixtures/models/CompositeKey'
```
Comment (iNaD) [inline test/feature/RootModules.spec.js]: Added a sample CompositeKey model for testing

File: test/feature/RootModules.spec.js [hunk 5]
```diff
@@ -53,6 +55,28 @@ describe('Root Modules', () => {
...
+      { user_id: 2, vote_id: 1, text: 'Jane' },
+      { user_id: 3, vote_id: 2, text: 'Johnny' }
+    ]
+
+    await store.dispatch('entities/create', { entity: 'compositeKeys', data })
+
+    await store.dispatch('entities/update', {
+      entity: 'compositeKeys',
+      data: { text: 'Christina' },
+      where: '3_2'
```
Comment (iNaD) [inline test/feature/RootModules.spec.js]: currently `Query` only supports internal id or condition based updates

File: test/feature/RootModules.spec.js [hunk 6]
```diff
@@ -53,6 +55,28 @@ describe('Root Modules', () => {
     expect(store.state.entities.users.data[2].name).toBe('Christina')
   })
 
+  it('can update data by action with composite primary key', async () => {
+    const store = createStore(entities)
+
+    const data = [
+      { user_id: 1, vote_id: 1, text: 'John' },
```
Comment (iNaD) [inline test/feature/RootModules.spec.js]: The composite key is based on `user_id` and `vote_id` values. Took this from the Docs about composite primary keys 😄

File: src/repo/Repo.ts (hunk 3 above)
Comment (kiaking) [reply src/repo/Repo.ts]: Yes it should cover the case of composite key with `this.entity.id`!

**Comment by kiaking** [comment]
This looks good! Thank you for the PR! I think forgot about update with composite key 🔥

**Pull request merged**